lib/urlwatch/handler.py

```diff
@@ -216,3 +238,61 @@ def parse_urls_txt(urls_txt):
 
     return jobs
 
+def parse_urls_yaml(urls_yaml):
+    jobs = []
+
+    # Security checks for shell jobs - only execute if the current UID
+    # is the same as the file/directory owner and only owner can write
+    allow_shelljobs = True
+    shelljob_errors = shelljob_security_checks(urls_yaml)
+    if shelljob_errors:
+        allow_shelljobs = False
+
+    with open(urls_yaml, 'r') as f:
+        yaml_jobs = yaml.load_all(f)
+        for job in yaml_jobs:
+            # entries with all parameters commented out
+            if job is None:
+                continue
+            # set all supported keys here, so we don't need to care later
+            for key in ['name', 'data']:
+                if not key in job:
+                    job[key] = None
+
+            if "url" in job:
+                jobs.append(UrlJob(job['url'], name=job['name'], data=job['data']))
+            elif "command" in job:
+                if allow_shelljobs:
+                    jobs.append(ShellJob(job['command'], name=job['name']))
+                else:
+                    print >>sys.stderr, '\n  SECURITY WARNING - Cannot run shell jobs:\n'
+                    for error in shelljob_errors:
+                        print >>sys.stderr, '    ', error
+                    print >>sys.stderr, '\n  Please remove shell jobs or fix these problems.\n'
+                    sys.exit(1)
+            else:
+                print >>sys.stderr, '\n  Entry has neither url nor command! Stopping!\n'
+                sys.exit(1)
+
+    return jobs
+
+def create_urls_yaml(jobs, urls_yaml):
+    with os.fdopen(os.open(urls_yaml, os.O_WRONLY | os.O_CREAT, 0o0644), 'w') as f:
```

Comment:
Because then the file could be group writable (depending on systems umask settings) and urlwatch would stop with a security warning, if the file contains ShellJobs.
I don't want to create a file automatically, that makes urlwatch stop working (because of permission problems).